### PR TITLE
Edit docs CI workflows to accommodate Docusaurus

### PR DESCRIPTION
**Pull request by ptgott**
- Copy the `gravitational/teleport` source into the `content` directory of a `gravitational/docs-website` clone and overwrite the `git-update` yarn script with a no-op command. With way, we can use the `gravitational/teleport` clone across multiple workflow steps (e.g., the prose linting step), and can identify a branch to pull using the `actions/checkout` defaults. Name the submodule directory `current` to match the expectations of the Docusaurus site.

- Remove the Vercel preview workflow. After the docs engine migration, we'll no longer need to deploy a preview to Vercel in order to show docs authors what their potential changes look like on a rendered docs site.

Also fixes spelling errors caught by a later version of cspell.

### CODE REVIEW

File: .github/workflows/doc-tests.yaml [hunk 1]
```diff
@@ -44,8 +44,14 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v4
         with:
-          repository: "gravitational/docs"
-          path: "docs"
+          repository: 'gravitational/teleport'
+          path: 'teleport'
```
Comment (zmb3) [inline .github/workflows/doc-tests.yaml:48]: Wouldn't it be equivalent to omit these two lines now?

**Review by zmb3 — APPROVED** [review]

**Comment by ptgott** [comment]
Converting to draft until we're closer to making the final switch.

File: .github/workflows/doc-tests.yaml [hunk 2]
```diff
@@ -80,30 +80,36 @@ jobs:
...
+        # 
         # The docs engine expects a config.json file at the root of the
         # gravitational/docs clone that associates directories with git
         # submodules. By default, these directories represent versioned branches
         # of gravitational/teleport. We override this in order to build only a
         # single version of the docs.
+        #
+        # We also replace data fetched from Sanity CMS with hardcoded JSON
+        # objects to remove the need to authenticate with Sanity. Each includes
+        # the minimal set of data required for docs builds to succeed.
```
Comment (fheinecke) [inline .github/workflows/doc-tests.yaml:95]: Please run this step through shellcheck. There's several issues that it will report.

**Comment by ptgott** [comment]
@fheinecke Checking to see if you have time to give this one another look. Thanks!

**Review by taraspos — APPROVED** [review]

**Comment by taraspos** [comment]
@ptgott we may want to remove vercel webhook as well:
- https://github.com/gravitational/teleport/blob/074d584b7a9447dfde15314455bf7e82f3604ccf/.github/workflows/update-docs-webhook.yaml#L20-L21

Unless we want to keep it for couple days and then possibly remove it next week

**Comment by ptgott** [comment]
> @ptgott we may want to remove vercel webhook as well:
> 
> * https://github.com/gravitational/teleport/blob/074d584b7a9447dfde15314455bf7e82f3604ccf/.github/workflows/update-docs-webhook.yaml#L20-L21
> 
> Unless we want to keep it for couple days and then possibly remove it next week

Thanks! I'll plan to remove this in a separate PR since we can run `Lint (Docs)` using `gravitational/docs-website` even before we route `/doc` traffic to the new site.

**Pull request merged**